nums/core/array/application.py

```diff
@@ -1114,3 +1114,31 @@ def atleast_3d(self, *arys):
             return res[0]
         else:
             return res
+
+    def hstack(self, tup):
```

Comment:
For all the stacking functions, could you add the `axis_block_size` parameter and pass it through to `concatenate`? This lets the user (at the `ArrayApplication` level) specify the block size for the axis along which the arrays are being concatenated. This parameter would not be provided at the numpy API level.